test/built-ins/Iterator/concat/many-arguments.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-iterator.concat
+description: >
+  Iterator.concat when called with many arguments.
+info: |
+  Iterator.concat ( ...items )
+
+  1. Let iterables be a new empty List.
+  2. For each element item of items, do
+    ...
+  3. Let closure be a new Abstract Closure with no parameters that captures iterables and performs the following steps when called:
+    a. For each Record iterable of iterables, do
+      ...
+    b. Return ReturnCompletion(undefined).
+  ...
+  6. Return gen.
+features: [iterator-sequencing]
+---*/
+
+let iterables = [
+  [/* empty */],
+  [1],
+  [2, 3],
+  [4, 5, 6],
+  [7, 8, 9, 10],
+];
+
+let iterator = Iterator.concat(...iterables);
+
+let array = [].concat(...iterables);
```

Comment:
As the champion of `Array.prototype.flat`, it breaks my heart to see this.

```suggestion
let array = iterables.flat();
```

Reply:
We try to avoid newer ES language features in tests if they're not the feature being tested - it's not a prohibition and many tests do it anyway, but we generally don't go out of our way to add them.